Java

```diff
@@ -1,44 +1,55 @@
 import java.util.Scanner; 
-import java.text.NumberFormat;
-class student{   
+    class Student{
+        String name;
+        String id;
+        int hours = 0;
+
+        public Student(String name, String id){
+            this.name = name;
+            this.id = id;
+        }
+
+        public void addHours(double hours){
+            this.hours += hours;
+        }
+
+        public double getHoursAsPercentage(double totalHours){
+            return hours * 100.0 / totalHours;
+        }
+    }
+class Main{   
 	
 	public static void main(String args[]) { 
 	
 		Scanner s = new Scanner(System.in); 
 		String newline = System.lineSeparator(); 
-		NumberFormat defaultFormat = NumberFormat.getPercentInstance();
-		defaultFormat.setMinimumFractionDigits(2);
-		
-		String id[] = new String[100]; 
-		String name[] = new String[100]; 
-		double time[] = new double[100]; 
-		String module;
-		double total = 0;
 	
-		System.out.println("Please enter the Module name:");
-		module = s.nextLine();
+	System.out.println("Please enter the Module name:");
+		String module = s.nextLine();
 		
 	System.out.println(newline);
 	
 		System.out.println(module+" Attendance Calculation");
 		
 	System.out.println(newline);
-		
-		System.out.println("Please enter the number of students:");
-		int n = s.nextInt();
-		
-		
-		for(int i=0; i<n; i++){
-		 System.out.println(newline);
-			System.out.print("Enter Student name # "+(i+1)+": "); 
-            name[i] = s.next(); 
-		
-	
-			 
-			System.out.print("Enter Student ID # "+(i+1)+": ");
-            id[i] = s.next();
-             
-		
+
+	System.out.println("Please enter the number of students:");
+	int n = s.nextInt();
+
+	Student[] students = new Student[n];
```

Comment:
That's good, better than hard-coding the maximum number of Students to 100 (as was done in the initial code)